src/PrestaShopBundle/Controller/Admin/TranslationsController.php

```diff
@@ -194,11 +196,22 @@ protected function findLanguageByLocale($locale)
     protected function getTranslationsCatalogue(Request $request)
     {
         $lang = $request->request->get('lang');
+        $type = $request->request->get('type');
 
         $translator = $this->container->get('translator');
 
         $locale = $this->langToLocale($lang);
+        
+        $catalogue = $this->get('ps.catalogue_factory')->createCatalogue($type, $locale);
+        
+        if ($catalogue instanceof MessageCatalogue) {
+            $translations = $catalogue->all();
+            ksort($translations);
+
+            return $translations;
+        }
 
+        // if type is not found, return all keys
         $finder = new Finder();
```

Comment:
ping @thierrymarianne, why do we need 'db' keys ? I was expecting a real Catalogue but it's a formatted associative array that is returned to front.